### DISCUSSION TITLE
[Feature] native mobile apps (iOS / Android)

### DISCUSSION THREAD

**Discussion by traktuner** [immich-app/immich#9788] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, the apps are built using Flutter for i(Pad)OS and Android.
Would it be possible, to develop both apps using native frameworks like Swift/SwiftUI and Kotlin?
This would result in a generalized OS-specific UI with probably much better performance on both platforms.

I have refrained from writing this feature request until now, as I understand the need for an open source project to use a universal framework. But maybe it makes more sense [now](https://immich.app/blog/2024/immich-core-team-goes-fulltime) :-)

Thank you very much!

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
This is not going to happen, as it would essentially mean rewriting the app from scratch twice and then maintaining those two versions. The flutter app works perfectly fine as it is. If we need OS-specific stuff flutter provides escape hatches for that.
